crypto/evp/evp_lib.c

```diff
@@ -637,14 +647,24 @@ int EVP_CIPHER_get_key_length(const EVP_CIPHER *cipher)
 
 int EVP_CIPHER_CTX_get_key_length(const EVP_CIPHER_CTX *ctx)
 {
-    int ok;
-    size_t v = ctx->key_len;
-    OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
-
-    params[0] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_KEYLEN, &v);
-    ok = evp_do_ciph_ctx_getparams(ctx->cipher, ctx->algctx, params);
-
-    return ok != 0 ? (int)v : EVP_CTRL_RET_UNSUPPORTED;
+    if (ctx->key_len < 0) {
+        /*-
+         * Casting away the const is annoying but required here.  We need to
+         * cache the result for performance reasons.
+         */
+        int *const len = &((EVP_CIPHER_CTX *)ctx)->key_len;
+        int ok;
+        OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
+
+        params[0] = OSSL_PARAM_construct_int(OSSL_CIPHER_PARAM_KEYLEN, len);
```

Comment:
Everywhere in the provider code this type is a size_t.

Reply:
I know but it's an `int`s in the structure & it isn't reasonable to change it.

The OSSL_PARAM helpers do the conversion from `int` to `size_t` for us.  It seemed clearer to not do the conversion here explicitly.

Reply:
changed to size_t